src/Interpreters/MonotonicityCheckVisitor.h

```diff
@@ -33,7 +37,23 @@ class MonotonicityCheckMatcher
         ASTIdentifier * identifier = nullptr;
         DataTypePtr arg_data_type = {};
 
-        void reject() { monotonicity.is_monotonic = false; }
+        Range range = Range::createWholeUniverse();
+
+        enum class ErrorCode
```

Comment:
I can instead throw exceptions. This visitor is used only in two places: `TreeOptimizer` and `MergeTreePartitionCompatibilityVerifier`. I can add a try catch on former if necessary.